lib/std/os.zig

```diff
@@ -3436,6 +3436,32 @@ pub fn munmap(memory: []align(mem.page_size) u8) void {
     }
 }
 
+pub usingnamespace if (!@hasDecl(system, "mremap")) struct { } else struct {
```

Comment:
why this `@hasDecl` check?

Reply:
Only some systems support mremap.

Reply:
`std.os.system` is pub so you can just check `@hasDecl(os.system, "mremap")` in `std.heap` and avoid this `usingnamespace`.

Reply:
> Only some systems support mremap.

mremap is Linux specific, but has been part of Linux ever since the other memory mapping functions, so it might
be preferable to simply check for `builtin.os == .linux`.

Reply:
@Snektron it looks like there may be some support elsewhere: https://www.freebsd.org/cgi/man.cgi?query=mremap&apropos=0&sektion=0&manpath=NetBSD+5.0&format=html

@vexu, it's unclear to me that is a better solution.  With `usingnamespace`, the `mremap` function is only defined in `os.zig` when the system supports it.  So you can do this:
```Zig
if (@hasDecl(std.os, "mremap")) {
    std.os.mremap(...)
} else {
    // do something else
}
```
But with your solution it sounds like all systems will declare the `mremap` function whether or not it is supported?  So rather than just checking if it exists, you'd have to use some other condition in the hopes that it matches the same condition that is used to determine which systems support `mremap`.

Reply:
All systems will also declare `openatWasi` even though it is only supported on `wasi`, I don't see checking if it exists in `std.os` being any easier than checking if it exists in `std.os.system`.

Reply:
Wouldn't you say it's more natural to check for the thing you're wanting to call directly rather than indirectly checking on a shared dependency?
```Zig
if (@hasDecl(std.os, "mremap")) std.os.mremap(...)
// VS
if (@hasDecl(std.os.system, "mremap")) std.os.mremap(...)
```
If I'm going to call `std.os.mremap`, it's more "stable" to ask if `std.os.mremap` exists instead of checking if another module has defined `mremap`.  This makes the caller agnostic to the relationship between `std.os.mremap` and `std.os.system.mremap`.

What's the advantage of doing it the other way?